ludwig/backend/horovod.py

```diff
@@ -69,4 +71,20 @@ def is_coordinator(self):
 
     @property
     def num_nodes(self) -> int:
+        if not self._horovod:
+            return 1
         return self._horovod.size()
+
+    @property
+    def num_cpus(self) -> int:
```

Comment:
The Horovod backend is not used with Horovod on Ray (that's what the Ray bacend does). The correct way to implement this would be to compute number of CPUs / GPUs on each worker independently, then allreduce the results:

```
cpu_count = psutil.cpu_count()
return hvd.allreduce(cpu_count)
```

Reply:
Got it, that makes a lot of sense!